### ISSUE TITLE
glm4v-9b-chat微调之后“打字机”问题变得很严重，每个问题都有重复输出和非中英文乱码。

### ISSUE THREAD

**Issue by Jade0321** [THUDM/GLM-4#428]
### System Info / 系統信息

用自定义数据进行了微调，数据格式如下：
{"messages": [{"role": "user", "content": "在电商领域中，客户和销售会进行对话，请你扮演销售的角色，对客户进行回复。\n\n电商对话：\n销售：方老师，您那笔505元的订单返现10元红包。麻烦您提供一下收款码给我，我这边给您扫码付款\n用户：[图片]\n\n    \n你的回复：\n", "image": "00012796.jpg"}, {"role": "assistant", "content": "好的，我尽快为您返现，请您关注收款信息。"}]}

![企业微信截图_17224793547473](https://github.com/user-attachments/assets/af090bc2-e96b-412e-aa75-b26b88a1cf4b)

### Who can help? / 谁可以帮助到您？

@wwewwt @Sengxian @

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [X] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

微调数据量在1500左右，微调命令：
OMP_NUM_THREADS=1 torchrun --standalone --nnodes=1 --nproc_per_node=8  finetune_vision.py  ./customer_data/  ./glm-4v-9b  configs/lora.yaml  

这是lora.yaml文件：
data_config:
  train_file: train-data-answer-all-label-final.jsonl
  val_file: valid-data-all-label-final-50.jsonl
  test_file: 
  num_proc: 1

combine: Trues
max_input_length: 1024
max_output_length: 1024

training_args:
  # see `transformers.Seq2SeqTrainingArguments`
  output_dir: ./output/train_0801_caption_09h28m
  max_steps: 3000
  # needed to be fit for the dataset
  learning_rate: 5e-4
  # settings for data loading
  per_device_train_batch_size: 1
  dataloader_num_workers: 16
  remove_unused_columns: false
  # settings for saving checkpoints
  save_strategy: steps
  save_steps: 500
  # settings for logging
  log_level: info
  logging_strategy: steps
  logging_steps: 10
  # settings for evaluation
  per_device_eval_batch_size: 2
  eval_strategy: steps
  eval_steps: 3000
  # settings for optimizer
  # adam_epsilon: 1e-6
  # uncomment the following line to detect nan or inf values
  # debug: underflow_overflow
  predict_with_generate: true
  # see `transformers.GenerationConfig`
  generation_config:
    max_new_tokens: 1024
  # set your absolute deepspeed path here
  # deepspeed: configs/ds_zero_3.json
peft_config:
  peft_type: LORA
  task_type: CAUSAL_LM
  r: 8
  lora_alpha: 32
  lora_dropout: 0.1
  target_modules: ["query_key_value"]

结果每条都是有重复输出，而且有很多诡异的字符，请各位大神帮忙看下，是不是这里没加文本起止符？
    @property
    def default_chat_template(self):
        """
        This template formats inputs in the standard ChatML format. See
        https://github.com/openai/openai-python/blob/main/chatml.md
        """
        return (
            "{% for message in messages %}"
            "{{'<|im_start|>' + message['role'] + '\n' + message['content'] + '<|im_end|>' + '\n'}}"
            "{% endfor %}"
            "{% if add_generation_prompt %}"
            "{{ '<|im_start|>assistant\n' }}"
            "{% endif %}"
        )

### Expected behavior / 期待表现

如果消除打字机现象？

**Comment by zRzRzRzRzRzRzR** [comment]
这个模板都不对啊，不是官方的微调代码吧

**Comment by zRzRzRzRzRzRzR** [comment]
数据格式，im_start 我们都没有这个special token

**Comment by Jade0321** [comment]
我微调的数据是：
![image](https://github.com/user-attachments/assets/d504bb3e-06fc-40bf-b083-38f27a3ef3c4)
跟readme当中的VQA示例应该是相同的：
![image](https://github.com/user-attachments/assets/1ba57e1d-b8d9-40a3-89d7-03aab963826e)

special token是通过打开下面图片中的函数tokenizer.apply_chat_template：
![image](https://github.com/user-attachments/assets/d90cf63b-a4e0-47e9-adbf-e7c5f5be85a5)
进到/usr/local/lib/miniconda3/envs/glm4/lib/python3.10/site-packages/transformers/tokenization_utils_base.py文件下面
有一个函数default_chat_template(self):
![image](https://github.com/user-attachments/assets/07b158ad-618b-40ac-93f3-02851c65f31d)
使用的是官方代码，请再帮忙看看

**Comment by zRzRzRzRzRzRzR** [comment]
这个没更新吧，官方的github中应该服写了这个呀
